sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DropTableExec.scala

```diff
@@ -26,13 +27,17 @@ import org.apache.spark.sql.connector.catalog.{Identifier, TableCatalog}
  * Physical plan node for dropping a table.
  */
 case class DropTableExec(
+    session: SparkSession,
     catalog: TableCatalog,
     ident: Identifier,
     ifExists: Boolean,
     purge: Boolean) extends V2CommandExec {
 
   override def run(): Seq[InternalRow] = {
     if (catalog.tableExists(ident)) {
+      val table = catalog.loadTable(ident)
```

Comment:
We don't need to look up the table again. It's already in `DropTable` and we can just pass it to `DropTableExec`